### PR TITLE
support option argument completion

### PR DESCRIPTION
**Pull request by tw4452852**
Signed-off-by: Tw <wei.tan@intel.com>

### CODE REVIEW

File: edit/completion/compl_getopt.go [hunk 1]
```diff
@@ -91,6 +92,22 @@ func complGetopt(fm *eval.Frame, elemsv, optsv, argsv interface{}) {
 		if s, ok := getStringField("arg-desc"); ok {
 			argdesc[opt] = s
 		}
+
+		getCallableField := func(k string) (eval.Callable, bool) {
```
Comment (xiaq) [inline edit/completion/compl_getopt.go]: Can you move the definition to just under `getStringField`?

File: edit/completion/compl_getopt.go [hunk 2]
```diff
@@ -91,6 +92,22 @@ func complGetopt(fm *eval.Frame, elemsv, optsv, argsv interface{}) {
...
+			if !ok {
+				return nil, false
+			}
+			if vb, ok := v.(eval.Callable); ok {
+				return vb, true
+			}
+			throwf("%s should be fn, got %s", k, vals.Kind(v))
+			panic("unreachable")
+		}
+		if f, ok := getCallableField("handler"); ok {
```
Comment (xiaq) [inline edit/completion/compl_getopt.go]: Can we call the field `completer` instead of `handler`?

File: edit/completion/compl_getopt.go [hunk 3]
```diff
@@ -141,6 +158,18 @@ func complGetopt(fm *eval.Frame, elemsv, optsv, argsv interface{}) {
 		out <- c
 	}
 
+	callHandler := func(fn eval.Callable, args []string) {
```
Comment (xiaq) [inline edit/completion/compl_getopt.go]: Can we call the function `callCompleter`?

File: edit/completion/compl_getopt.go (hunk 1 above)
Comment (tw4452852) [reply edit/completion/compl_getopt.go]: I also put all three functions (`getCallableField`,`getCallableField` and `getBoolField`) together, hope to make it more clear.

**Comment by tw4452852** [comment]
@xiaq Of course, I have updated reference manual.

**Pull request merged**